kotlin/generate-geodatabase/src/main/java/com/esri/arcgisruntime/sample/generategeodatabase/MainActivity.kt

```diff
@@ -0,0 +1,164 @@
+/* Copyright 2020 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.esri.arcgisruntime.sample.generategeodatabase
+
+import android.graphics.Color
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.concurrent.Job
+import com.esri.arcgisruntime.data.TileCache
+import com.esri.arcgisruntime.layers.ArcGISTiledLayer
+import com.esri.arcgisruntime.layers.FeatureLayer
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.view.Graphic
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import com.esri.arcgisruntime.tasks.geodatabase.GeodatabaseSyncTask
+import kotlinx.android.synthetic.main.activity_main.*
+import java.io.File
+
+class MainActivity : AppCompatActivity() {
+
+  private val TAG = MainActivity::class.java.simpleName
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+    // use local tile package for the base map
+    val sanFrancisco =
+      TileCache(getExternalFilesDir(null).toString() + getString(R.string.san_francisco_tpk))
+    val tiledLayer = ArcGISTiledLayer(sanFrancisco)
+
+    // add the map and graphics overlay to the map view
+    mapView.apply {
+      // create a map with the tile package basemap
+      this.map = ArcGISMap(Basemap(tiledLayer))
+      // create a graphics overlay to display the boundaries
+      graphicsOverlays.add(GraphicsOverlay())
+    }
+
+    // generate geodatabase when the button is clicked
+    genGeodatabaseButton.setOnClickListener {
+      generateAndDisplayGeodatabase()
+    }
+  }
+
+  /**
+   * Creates a GenerateGeodatabaseJob and displays its results on the map.
+   */
+  private fun generateAndDisplayGeodatabase() {
+    // create a geodatabase sync task and load it
+    val geodatabaseSyncTask =
+      GeodatabaseSyncTask(getString(R.string.wildfire_sync)).apply { loadAsync() }
+
+    geodatabaseSyncTask.addDoneLoadingListener onTaskLoaded@{
```

Comment:
This is because of the early returning later on; there is a nested `addDoneLoadingListener` which means that writing `return@addDoneLoadingListener` on line 127 is ambiguous. I put the annotation here because I don't need to return to this scope and giving it an annotation removes the ambiguity. I was a little unsure about this so I'd really like to know what you think

Reply:
Oh I see, makes sense. If I can convince you on un-nesting geodatabase work in the job's done listener, this issue would go away too :)